sched/wqueue/kwork_notifier.c

```diff
@@ -143,24 +133,16 @@ static FAR struct work_notifier_entry_s *work_notifier_find(int16_t key)
  *
  ****************************************************************************/
 
-static int16_t work_notifier_key(void)
+static uint32_t work_notifier_key(void)
 {
-  int16_t key;
-
-  /* Loop until a unique key is generated.  Range 1-INT16_MAX. */
+  static uint32_t notifier_key;
 
-  do
+  if (++notifier_key == 0)
```

Comment:
Would it be better to make it obvious as to intent?

```suggestion
  ++notifier_key;
  if (notifier_key == 0) 
    {
      notifier_key = 1;
    }
```

Reply:
Yes, it looks more clearer, please review again!